common/ManagementTestShared/Redesign/ManagementRecordedTestBase.cs

```diff
@@ -0,0 +1,163 @@
+        {
+            SessionEnvironment = new TEnvironment();
+            SessionEnvironment.Mode = Mode;
+        }
+
+        protected ManagementRecordedTestBase(bool isAsync, RecordedTestMode mode) : base(isAsync, mode)
+        {
+            SessionEnvironment = new TEnvironment();
+            SessionEnvironment.Mode = Mode;
+        }
+
+        private AzureResourceManagerClient GetCleanupClient()
+        {
+            if (Mode != RecordedTestMode.Playback)
+            {
+                return new AzureResourceManagerClient(
+                        TestEnvironment.SubscriptionId,
+                        TestEnvironment.Credential,
+                        new AzureResourceManagerClientOptions());
+            }
+            return null;
+        }
+
+        protected AzureResourceManagerClient GetArmClient()
+        {
+            var options = InstrumentClientOptions(new AzureResourceManagerClientOptions());
+            options.AddPolicy(CleanupPolicy, HttpPipelinePosition.PerCall);
+
+            return CreateClient<AzureResourceManagerClient>(
+                TestEnvironment.SubscriptionId,
+                TestEnvironment.Credential,
+                options);
+        }
+
+        [SetUp]
+        protected void Setup()
+        {
+            _cleanupClient ??= GetCleanupClient();
+        }
+
+        [TearDown]
+        protected void CleanupResourceGroups()
+        {
+            if (Mode != RecordedTestMode.Playback)
+            {
+                Parallel.ForEach(CleanupPolicy.ResourceGroupsCreated, resourceGroup =>
+                {
+                    _cleanupClient.GetResourceGroupOperations(TestEnvironment.SubscriptionId, resourceGroup).StartDelete();
+                });
+            }
+        }
+
+        private void StartSessionRecording()
+        {
+            // Only create test recordings for the latest version of the service
+            TestContext.TestAdapter test = TestContext.CurrentContext.Test;
+            if (Mode != RecordedTestMode.Live &&
+                test.Properties.ContainsKey("SkipRecordings"))
+            {
+                throw new IgnoreException((string)test.Properties.Get("SkipRecordings"));
+            }
+            SessionRecording = new TestRecording(Mode, GetSessionFilePath(), Sanitizer, Matcher);
+            SessionEnvironment.SetRecording(SessionRecording);
+            ValidateClientInstrumentation = SessionRecording.HasRequests;
+        }
+
+        protected void StopSessionRecording()
+        {
+            if (ValidateClientInstrumentation)
+            {
+                throw new InvalidOperationException("The test didn't instrument any clients but had recordings. Please call InstrumentClient for the client being recorded.");
+            }
+
+            SessionRecording?.Dispose(true);
+            GlobalClient = null;
+        }
+
+        [OneTimeSetUp]
+        public void OneTimeSetUp()
+        {
+            if (!HasOneTimeSetup())
```

Comment:
We might be able to base this check on the session being empty.

Reply:
Unless its the first recording right?

Reply:
If the check is here to prevent an empty recording file we can check it before writing the file.